src/daft-sql/src/functions.rs

```diff
@@ -274,9 +282,9 @@ impl<'a> SQLPlanner<'a> {
                 unsupported_sql_err!("subquery function argument")
             }
             sqlparser::ast::FunctionArguments::List(args) => {
-                if args.duplicate_treatment.is_some() {
-                    unsupported_sql_err!("function argument with duplicate treatment");
-                }
+                if matches!(args.duplicate_treatment, Some(DuplicateTreatment::Distinct)) {
+                    fn_match = get(&SQL_FUNCTIONS, "count_distinct")?;
+                };
```

Comment:
Also check if the original function is a count. DISTINCT is not supported for other ops

Reply:
Updated.